src/traits.rs

```diff
@@ -1,4 +1,63 @@
-use crate::{IsoWeek, Weekday};
+use core::ops::{Add, RangeInclusive};
+
+use crate::{Duration, IsoWeek, Weekday};
+
+/// The common set of methods for a week component.
+pub trait Weeklike: Sized {
+    /// An associated type, representing a day within the context of a week.
+    type Day: Add<Duration, Output = Self::Day>;
+
+    /// Returns a date representing the start of the week.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use chrono::{NaiveDate, Weekday, Weeklike};
+    ///
+    /// let date = NaiveDate::from_ymd(2022, 4, 18);
+    /// let week = date.week(Weekday::Mon);
+    /// assert!(week.start_day() <= date);
+    /// ```
+    fn start_day(&self) -> Self::Day;
+
+    /// Returns a date representing the end of the week.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use chrono::{NaiveDate, Weekday, Weeklike};
+    ///
+    /// let date = NaiveDate::from_ymd(2022, 4, 18);
+    /// let week = date.week(Weekday::Mon);
+    /// assert!(week.end_day() >= date);
+    /// ```
+    #[inline]
+    fn end_day(&self) -> Self::Day {
+        self.start_day() + Duration::days(6)
+    }
+
+    /// Returns a [`RangeInclusive<T>`] representing the whole week bounded by
+    /// [start_day()][start_day] and [end_day()][end_day] functions.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use chrono::{NaiveDate, Weekday, Weeklike};
+    ///
+    /// let date = NaiveDate::from_ymd(2022, 4, 18);
+    /// let week = date.week(Weekday::Mon);
+    /// let range = week.range();
+    /// assert_eq!(range.start(), &week.start_day());
+    /// assert_eq!(range.end(), &week.end_day());
+    /// ```
+    ///
+    /// [start_day]: ./trait.Weeklike.html#tymethod.start_day
+    /// [end_day]: ./trait.Weeklike.html#method.end_day
+    #[inline]
+    fn range(&self) -> RangeInclusive<Self::Day> {
```

Comment:
I think I'd prefer to call this `days()` to make it clear what the range's `Item` is. What do you think?